src/main/java/com/puppycrawl/tools/checkstyle/checks/DescendantTokenCheck.java

```diff
@@ -155,8 +157,12 @@ public class DescendantTokenCheck extends AbstractCheck {
      * the sum of the individual token counts.
      */
     private boolean sumTokenCounts;
-    /** Specify set of tokens with limited occurrences as descendants. */
-    @XdocsPropertyType(PropertyType.TOKEN_ARRAY)
+    /**
+     * Specify set of
+     * <a href="../../apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">tokens</a>
+     * with limited occurrences as descendants.
+     */
+    @XdocsPropertyType(PropertyType.STRING_ARRAY)
```

Comment:
I am still reading prior discussion https://github.com/checkstyle/checkstyle/pull/13925#discussion_r1368922550 , but this change doesn't seem valid.
Property is an array of tokens.

Reply:
> Because it is failing this condition: if (SiteUtil.TOKENS.equals(propertyName))
> One approach would be to have a specific use-case of limitedTokens, and create a condition to match it with the original xdocs. 

We should not have to, and really avoid, creating a custom override list of types for specific checks and properties. Everything is documented in class that this is a token list, so we should be using that.

It looks to me that code needs to be re-written to take actual type from `XdocsPropertyType`, which I am sure I mentioned in some other issue and then just have somewhat special handling of token lists as there is no pre-defined acceptable/required list like with natural `tokens` property.

Reply:
That can be may be substantiated by @Romani. But, i will just add some more context to it. If we refer `DescendantTokenCheck.xml`, it says limitedTokens is of `String[]`.
```
            <property default-value=""
                      name="limitedTokens"
                      type="java.lang.String[]"
                      validation-type="tokenTypesSet">
               <description>Specify set of tokens with limited occurrences as descendants.</description>
            </property>
```

Reply:
Then it sounds like a bug in the metadata as the property is a representation of tokens, and no 3rd party utility will know this since we say its just a simple string array.

Reply:
Even if we are suppose to take the type from `validation-type`, which is still slightly confusing, especially since we have no documentation of the metadata and what any of it really means. It doesn't change that this is a token array, and how will metadata know to use the `validation-type` anyways.

This doesn't seem to be making much sense to me.

Reply:
https://github.com/checkstyle/checkstyle/blob/8896be8cd4e90c59fdba8e048000b0f3472245c9/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L374

https://github.com/checkstyle/checkstyle/blob/8896be8cd4e90c59fdba8e048000b0f3472245c9/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java#L107-L113

It looks like there is another area that is just a custom hack list which should be using `XdocsPropertyType`.

All these partial GSoC fixes are just causing a mess with the code.

Reply:
Sure, i am leaning with the go. Let me know how to proceed from here.

Reply:
How to proceed is at https://github.com/checkstyle/checkstyle/pull/13925#discussion_r1375306101 .

After you do that, I recommend to create an issue for https://github.com/checkstyle/checkstyle/pull/13925#discussion_r1375308387 and work on it also since it will be similar areas to the 2.

Reply:
This is very confusing update , I agree, and kept it unresolved to share through process.

Confusion is that we use this annotation on fields but it actually should represent type of setters. We naively thought that all our properties has fully aligned setter and field. But here are exceptions. 
To avoid doing breaking change on setter type we should put annotation with different type on class field.

I am ok to create breaking  compatibility issue  and resolve it separately. But for now it is like what we have in this PR.
Do you want us to create this issue ?

Reply:
> Confusion is that we use this annotation on fields but it actually should represent type of setters
> We naively thought that all our properties has fully aligned setter and field

There is too much baggage here, but basically this has been known since the beginning. `XdocsPagesTest` handles identifying types as we have created since the beginning. I may be biased since I created it and worked it out, but it has stood the test of time for this long.

You can't go by setter alone, and you can't go by field alone. There won't always be a field, and due to limitations placed by ourselves on allowing extension of the bean utility, setters are purposely restricted to simple types. We have shown both of those things in this PR. `aliasList` has no field, and this class' setter is a simple string array for a token array type.

> But here are exceptions

To me there are very few exceptions. I pointed out now how a custom override list can be maintained through our annotations.

If you want to see how types are identified, look at https://github.com/checkstyle/checkstyle/blob/bd3a9bb9ec9cf804f628171c2e7557b0afbcfd93/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1369 and the surrounding code.

The field is the primary authority. If it doesn't exist, we go against the setter. See above on why we can't solely rely on one or the other.

> To avoid doing breaking change on setter type we should put annotation with different type on class field.

I am not sure I understand, but I am not sure it really matters. Annotation aren't breaking compatibility. Changing setter type won't really break compatibility as long as values in configs can remain inputted the same.

> Do you want us to create this issue ?

You can go about this as you want, assuming we still need a consensus, but we need to start planning better and providing actual fixes and less overrides.

If you want my opinion, we should start fixing issues where it has been pointed out things are wrong as it will just come up and block future migration work.

I don't see the change in this PR being the way.

Reply:
@rnveach, i have reverted back to keep changes for token_array. though, my CI is failing, pointing 
`<td><i><a href="property_types.html#subset_of_tokens_TokenTypes">../../property_types.html#subset_of_tokens_TokenTypes</a>: doesn't exist.</i></td></tr>`. 

Even if i didn't add any logic in this recent push, simply relying on some other fix that was merged which also had a failed macros. I am currently looking into ti. But, your review would be insightful.